src/status_im/ui/screens/wallet/send/events.cljs

```diff
@@ -266,191 +19,25 @@
+            (assoc-in [:wallet :send-transaction :symbol] symbol)
+            (assoc-in [:wallet :send-transaction :amount] nil)
+            (assoc-in [:wallet :send-transaction :amount-text] nil)
+            (assoc-in [:wallet :send-transaction :asset-error] nil))}))
 
 (handlers/register-handler-fx
  :wallet.ui/sign-transaction-button-clicked
  (fn [{:keys [db] :as cofx} _]
-   (let [keycard-account? (boolean (get-in db [:account/account :keycard-instance-uid]))]
-     (if keycard-account?
-       (send-keycard-transaction cofx)
-       {:db (assoc-in db [:wallet :send-transaction :show-password-input?] true)}))))
-
-(fx/defn keycard-hash-message
-  [_ data typed?]
-  (if typed?
-    {::hash-typed-data {:data         data
-                        :on-completed #(re-frame/dispatch [::hash-message-completed (types/json->clj %)])}}
-    {::hash-message {:message      (ethereum/naked-address data)
-                     :on-completed #(re-frame/dispatch [::hash-message-completed (types/json->clj %)])}}))
-
-(handlers/register-handler-fx
- :wallet.ui/sign-message-button-clicked
- (fn [{:keys [db] :as cofx} [_ typed? screen-params password-error-cb]]
-   (let [{:keys [data from password]} screen-params
-         keycard-account? (boolean (get-in db [:account/account :keycard-instance-uid]))
-         modal? (= (:view-id db) :wallet-sign-message-modal)]
-     (if keycard-account?
-       (fx/merge cofx
-                 {:db (assoc-in db [:navigation/screen-params :wallet-send-modal-stack :modal?] modal?)}
-                 (keycard-hash-message data typed?))
-       (if typed?
-         {::sign-typed-data {:data     data
-                             :password password
-                             :account  from
-                             :on-completed #(re-frame/dispatch [::sign-message-completed
-                                                                screen-params
-                                                                (types/json->clj %)
-                                                                password-error-cb])}}
-         {::sign-message {:params       {:data     data
-                                         :password (security/safe-unmask-data password)
-                                         :account  from}
-                          :on-completed #(re-frame/dispatch [::sign-message-completed
-                                                             screen-params
-                                                             (types/json->clj %)
-                                                             password-error-cb])}})))))
+   (let [{:keys [to symbol amount]} (get-in cofx [:db :wallet :send-transaction])
+         {:keys [symbol address]} (tokens/asset-for (:wallet/all-tokens db) (keyword (:chain db)) symbol)
+         amount-hex (str "0x" (abi-spec/number-to-hex amount))
+         to-norm (ethereum/normalized-address to)]
+     (signing/sign cofx {:tx-obj    (if (= symbol :ETH)
```

Comment:
Similar code for tx-obj construction is used here (Yielding protocol impl for Send command): https://github.com/status-im/status-react/blob/401969db27f68813ecf6960a0768ab746ac5ba6b/src/status_im/chat/commands/impl/transactions.cljs#L309. Maybe makes sense to extract it into a separate fn.